lib/dml/mongoose/index.js

```diff
@@ -67,20 +67,28 @@ MongooseDB.prototype.connect = function(db) {
   
   // Construct the authentication part of the connection string.
   authenticationString = dbUser && dbPass ? dbUser + ':' + dbPass + '@' : '';
-  
+
+  var connectionString = '';
+
   // Check if a MongoDB replicaset array has been specified.
   if (dbReplicaset && Array.isArray(dbReplicaset) && dbReplicaset.length !== 0) {
     // The replicaset should contain an array of hosts and ports   
-    this.conn = mongoose.createConnection('mongodb://' + authenticationString + dbReplicaset.join(',') + '/' + dbName, options);
+    connectionString = 'mongodb://' + authenticationString + dbReplicaset.join(',') + '/' + dbName
   } else {
     // Get the host and port number from the configuration.
     dbHost = configuration.getConfig('dbHost');  
     dbPort = configuration.getConfig('dbPort'); 
 
     portString = dbPort ? ':' + dbPort : '';
 
-    this.conn = mongoose.createConnection('mongodb://' + authenticationString + dbHost + portString + '/' + dbName, options);
+    connectionString = 'mongodb://' + authenticationString + dbHost + portString + '/' + dbName;
   }
+
+  if(typeof configuration.getConfig('dbAuthSource') !== 'undefined' && configuration.getConfig('dbAuthSource') !== '' ){
```

Comment:
`configuration.getConfig('dbAuthSource')` wouldn't ever return `undefined` because it falls back to an empty string: https://github.com/adaptlearning/adapt_authoring/blob/master/lib/configuration.js#L142.